amethyst_renderer/src/scene.rs

```diff
@@ -0,0 +1,63 @@
+//! A fully renderable scene.
+//!
+//! # Example
+//!
+//! ```rust
+//! let mut scene = Scene::default();
+//! scene.add_light(PointLight::default());
+//! scene.add_mesh(sphere, material);
+//! ```
+
+use cam::Camera;
+use fnv::FnvHashMap as HashMap;
+use light::Light;
+use mesh::Mesh;
+use mtl::Material;
+use std::collections::hash_map::Values;
+
+/// Immutable slice iterator of lights.
+pub type LightIter<'l> = Values<'l, String, Light>;
+
+/// Immutable slice iterator of meshes.
+pub type MeshIter<'m> = Values<'m, String, Mesh>;
+
+/// Collection of lights and meshes to render.
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct Scene {
+    camera: Vec<Camera>,
+    lights: HashMap<String, Light>,
```

Comment:
@ebkalderon Maybe you could use a `Vec` and a `FnvHashMap` which maps `String` -> `usize`? Would be more cache efficient and still ergonomic.